numba/misc/numba_sysinfo.py

```diff
@@ -0,0 +1,639 @@
+_machine = 'Machine'
+_cpu_name, _cpu_count = 'CPU Name', 'CPU Count'
+_cpus_allowed, _cpus_list = 'CPUs Allowed', 'List CPUs Allowed'
+_cpu_features = 'CPU Features'
+_cfs_quota, _cfs_period = 'CFS Quota', 'CFS Period',
+_cfs_restrict = 'CFS Restriction'
+_mem_total, _mem_available = 'Mem Total', 'Mem Available'
+# OS info
+_platform_name, _platform_release = 'Platform Name', 'Platform Release'
+_os_name, _os_version = 'OS Name', 'OS Version'
+_os_spec_version = 'OS Specific Version'
+_libc_version = 'Libc Version'
+# Python info
+_python_comp = 'Python Compiler'
+_python_impl = 'Python Implementation'
+_python_version = 'Python Version'
+_python_locale = 'Python Locale'
+# LLVM info
+_llvm_version = 'LLVM Version'
+# CUDA info
+_cu_dev_init = 'CUDA Device Init'
+_cu_drv_ver = 'CUDA Driver Version'
+_cu_detect_out, _cu_lib_test = 'CUDA Detect Output', 'CUDA Lib Test'
+# ROC information
+_roc_available, _roc_toolchains = 'ROC Available', 'ROC Toolchains'
+_hsa_agents_count, _hsa_agents = 'HSA Agents Count', 'HSA Agents'
+_hsa_gpus_count, _hsa_gpus = 'HSA Discrete GPUs Count', 'HSA Discrete GPUs'
+# SVML info
+_svml_state, _svml_loaded = 'SVML State', 'SVML Lib Loaded'
+_llvm_svml_patched = 'LLVM SVML Patched'
+_svml_operational = 'SVML Operational'
+# Threading layer info
+_tbb_thread, _tbb_error = 'TBB Threading', 'TBB Threading Error'
+_openmp_thread, _openmp_error = 'OpenMP Threading', 'OpenMP Threading Error'
+_openmp_vendor = 'OpenMP vendor'
+_wkq_thread, _wkq_error = 'Workqueue Threading', 'Workqueue Threading Error'
+# Numba info
+_numba_env_vars = 'Numba Env Vars'
+# Conda info
+_conda_build_ver, _conda_env_ver = 'Conda Build', 'Conda Env'
+_conda_platform, _conda_python_ver = 'Conda Platform', 'Conda Python Version'
+_conda_root_writable = 'Conda Root Writable'
+# Packages info
+_inst_pkg = 'Installed Packages'
+# Psutil info
+_psutil = 'Psutil Available'
+# Errors and warnings
+_errors = 'Errors'
+_warnings = 'Warnings'
+
+# Error and warning log
+_error_log = []
+_warning_log = []
+
+
+def get_os_spec_info(os_name):
+    # Linux man page for `/proc`:
+    # http://man7.org/linux/man-pages/man5/proc.5.html
+
+    # Windows documentation for `wmic OS`:
+    # https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/cim-operatingsystem
+
+    # MacOS man page for `sysctl`:
+    # https://www.unix.com/man-page/osx/3/sysctl/
+    # MacOS man page for `vm_stat`:
+    # https://www.unix.com/man-page/osx/1/vm_stat/
+
+    class CmdBufferOut(tuple):
+        buffer_output_flag = True
+
+    shell_params = {
+        'Linux': {
+            'cmd': (
+                CmdBufferOut(('tail', '-vn', '+1',
+                              '/sys/fs/cgroup/cpuacct/cpu.cfs_quota_us')),
+                CmdBufferOut(('tail', '-vn', '+1',
+                              '/sys/fs/cgroup/cpuacct/cpu.cfs_period_us')),
+            ),
+            'cmd_optional': (
+                ('cat', '/proc/meminfo'),
+                ('cat', '/proc/self/status'),
```

Comment:
Whilst appreciably convenient, I'd rather see the use of direct file reading from Python opposed to spawning processes to run commands in a shell, particularly on OS which use files for config.

Reply:
Fixed with 833cafc. Was it a security concern?

Reply:
Thanks. My concern was use of subprocess to read a text file, less to go wrong with a direct read.